algorithms/CPlusPlus/Trees/min-heap.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+ Implementation of the min-heap using an array.
+
+ Min-Heap is a Binary tree where each child nodes value is greater 
+ than that of their parent. It is used when you need quick access to 
+ the smallest number in the array.
+*/
+
+#include <iostream>
+#include <stdexcept>
+
+class Heap{
+	public:
+		void insert(const int& data);
+		void removemin();
+		void print();
+		Heap();
+		~Heap();
+
+	private:
+		// size - number of elements in the array
+		// capacity - number of elements the array can store
+		int size, capacity;
+		int *items;
+
+		int parent(unsigned index) const;
+		int child(unsigned index) const;
+
+		void heapifyUP(unsigned index);
+		void heapifyDOWN(unsigned index);
+		void growArray();
+
+};
+
+Heap::~Heap(){
+	delete items;
+	items = nullptr;
+}
+
+Heap::Heap(): size(0), capacity(2){
+	items = new int[capacity + 1];
+}
+
+int Heap::parent(unsigned index) const{ return index/2; }
+
+void Heap::growArray(){
+	/**
+	  * Inorder to accommodate more numbers in the array
+	  * Allocate memory in the heap 
+	  * Double the size of the old array up to (capacity*2) + 1
+	  * Copy the elements of the previous array into the new array
+	  */
+
+	// Doubling the size of the array
+	int* new_Array = new int[(capacity * 2) + 1];
+	// Copying the elements of the old array to the new array
+	for(int i=1; i<=size; i++){ new_Array[i] = items[i]; }
+	// Doubling the capacity
+	capacity *= 2;
+	// delete the items inorder to avoid any memory leak
+	delete items;
```

Comment:
```suggestion
	delete[] items;
```

Reply:
looks like you missed the deallocation in `void Heap::growArray()` function

Reply:
All done!!